src/main/java/hudson/plugin/versioncolumn/JVMVersionMonitor.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * The MIT License
+ * 
+ * Copyright (c) 2017-, Baptiste Mathus
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.plugin.versioncolumn;
+
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Computer;
+import hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor;
+import hudson.node_monitors.NodeMonitor;
+import hudson.remoting.Callable;
+import hudson.slaves.OfflineCause;
+import jenkins.security.MasterToSlaveCallable;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+public class JVMVersionMonitor extends NodeMonitor {
```

Comment:
Having written also https://github.com/batmat/inodes-monitor-plugin/ in the past, I can only agree! This should be easier.